Content.Client/Lobby/UI/LobbyCharacterPreviewPanel.cs

```diff
@@ -165,5 +165,15 @@ public static void GiveDummyJobClothes(EntityUid dummy, HumanoidCharacterProfile
                 }
             }
         }
+
+        protected override void FrameUpdate(FrameEventArgs args)
+        {
+            if (_entityManager.Deleted(_previewDummy))
+            {
```

Comment:
Checking every frame for the entity deletion seems kinda suss

Reply:
Yeah, I agree it's not great; the first commit ([92c56f7](https://github.com/space-wizards/space-station-14/pull/16382/commits/92c56f7cf8693873cfc2d19acced961e5aac6f78)) used a callback, but it was firing for all entities, even when in-game and the UI wasn't visible. Is there a better way to detect when this guy was deleted (or, to prevent it being deleted, in the first place?)

Reply:
Does adding the UpdateUI call anywhere in this file the entity deletion is called no worky?

Reply:
Yeah, but the entity is deleted by a system that this UI has no visibility to. [92c56f7](https://github.com/space-wizards/space-station-14/pull/16382/commits/92c56f7cf8693873cfc2d19acced961e5aac6f78) does work, but it always fires for all entities, even when the round is running. I reasoned that checking for deletion only when the UI was displayed would be better for game performance, but happy to apply that change instead, if you think that's better?

Reply:
Yeah that commit is significantly worse as it slows down deletion in the entire game rather than just checking every frame in lobby.